clang/lib/Driver/ToolChains/ClassicFlang.cpp

```diff
@@ -82,12 +82,18 @@ void ClassicFlang::ConstructJob(Compilation &C, const JobAction &JA,
     Stem = C.getDriver().GetTemporaryPath("", "");
   } else {
     OutFile = Output.getFilename();
+    if (OutFile == "-.ll")
+      OutFile = "tmp";
     Stem = llvm::sys::path::filename(OutFile);
     llvm::sys::path::replace_extension(Stem, "");
   }
 
   // Add input file name to the compilation line
-  UpperCmdArgs.push_back(Input.getBaseInput());
+  const char *InputFile = Input.getBaseInput();
+  if (strcmp(InputFile, "-") == 0)
+    UpperCmdArgs.push_back("/dev/stdin");
```

Comment:
I guess this wouldn't work on Windows? @kaadam @xoviat @isuruf Any thought?

Reply:
@bryanpkc Since i cannot test it on windows have You got any idea how to do it there?

Reply:
You can't to it on Windows. At least not with the 'path to a device file' approach, since Windows does not share the UNIX-specific 'everything is a file' philosophy. I guess the easiest way is to add the following preprocessor check:
```
#if !defined(_WIN32) && !defined(_WIN64)
...
#endif
```

Reply:
According to [StackOverflow](https://stackoverflow.com/questions/313111/is-there-a-dev-null-on-windows), on Windows, Command Prompt recognizes a pseudo-file named `NUL`, and PowerShell recognizes `$null`. While they work as the output of a pipe, I don't know whether they can be used as an input to a pipe.